godot-ffi/src/godot_ffi.rs

```diff
@@ -19,74 +19,56 @@ use std::marker::PhantomData;
 /// must properly initialize and clean up values given the [`PtrcallType`] provided by the caller.
 #[doc(hidden)] // shows up in implementors otherwise
 pub unsafe trait GodotFfi {
+    #[doc(hidden)]
     fn variant_type() -> sys::VariantType;
+
+    #[doc(hidden)]
     fn default_param_metadata() -> sys::GDExtensionClassMethodArgumentMetadata {
         sys::GDEXTENSION_METHOD_ARGUMENT_METADATA_NONE
     }
 
     /// Construct from Godot opaque pointer.
     ///
+    /// This will increment reference counts if the type is reference counted. If you need to avoid this then a `borrow_sys` method should
+    /// usually be used. Which is a method that takes a sys-pointer and returns it as a `&Self` reference. This must be manually implemented for
+    /// each relevant type as not all types can be borrowed like this.
```

Comment:
```suggestion
    /// This will increment reference counts if the type is reference counted. If you need to avoid this, then a `borrow_sys` associated function should
    /// usually be used. That function takes a sys-pointer and returns it as a `&Self` reference. This must be manually implemented for
    /// each relevant type, as not all types can be borrowed like this.
```